src/stcal/ramp_fitting/utils.py

```diff
@@ -1368,24 +1405,51 @@ def dq_compress_sect(ramp_data, num_int, gdq_sect, pixeldq_sect):
     ngroups, nrows, ncols = gdq_sect.shape
 
     # If all groups are set to DO_NOT_USE, mark as DO_NOT_USE.
-    dnu = np.zeros(gdq_sect.shape, dtype=np.uint32)
-    dnu[np.where(np.bitwise_and(gdq_sect, dnu_flag))] = 1
-    dnu_sum = dnu.sum(axis=0)
-    all_dnu = np.where(dnu_sum == ngroups)
-    pixeldq_sect[all_dnu] = np.bitwise_or(pixeldq_sect[all_dnu], dnu_flag)
+    is_flag_set_for_all_groups(pixeldq_sect, gdq_sect, dnu_flag)
 
-    # If saturation or a jump occures mark the appropriate flag.
-    sat_loc_r = np.bitwise_and(gdq_sect, sat_flag)
-    sat_loc_im = np.where(sat_loc_r.sum(axis=0) > 0)
-    pixeldq_sect[sat_loc_im] = np.bitwise_or(pixeldq_sect[sat_loc_im], sat_flag)
+    # If all groups are set to SATURATED, mark as SATURATED.
+    # A group 0 marked as saturated implies all groups are saturated.
+    gdq0_sat = np.bitwise_and(gdq_sect[0], sat_flag)
+    pixeldq_sect[gdq0_sat != 0] = np.bitwise_or(
+        pixeldq_sect[gdq0_sat != 0], sat_flag)
 
+    # If jump occures mark the appropriate flag.
     cr_loc_r = np.bitwise_and(gdq_sect, jump_flag)
     cr_loc_im = np.where(cr_loc_r.sum(axis=0) > 0)
     pixeldq_sect[cr_loc_im] = np.bitwise_or(pixeldq_sect[cr_loc_im], jump_flag)
 
     return pixeldq_sect
 
 
+def is_flag_set_for_all_groups(pixeldq_sect, gdq_sect, flag):
+    """
```

Comment:
Like the is_flag_set for_all_ints( ) function above, a bit more explanation in the docs would be helpful for this function.